### PR TITLE
Let ana_scans fail when any command fails

### PR DESCRIPTION
**Pull request by lmoureaux**
## Description
Addresses #28 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and Context

Needed for unit testing, see #27 and #28

## How Has This Been Tested?

Tested on Travis with `--anaType=scurve`. Worked fine both in case of success and in case of failure of the main analysis (output logs [here](https://travis-ci.org/lmoureaux/gem-plotting-tools/jobs/268001263) (success), and [here](https://travis-ci.org/lmoureaux/gem-plotting-tools/jobs/267992000) (analysis fails)).

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (not in this PR though).
- [x] All new and existing tests passed (not in this PR though).

### CODE REVIEW

File: ana_scans.py [hunk 1]
```diff
@@ -125,15 +125,22 @@ def launchAnaArgs(anaType, cName, cType, scandate, scandatetrim=None, ztrim=4.0,
   try:
     log = file("%s/anaLog.log"%(dirPath),"w")
  
-    #runCommand(cmd,log)
-    runCommand(cmd)
+    #returncode = runCommand(cmd,log)
+    returncode = runCommand(cmd)
```
Comment (jsturdy) [inline ana_scans.py]: Why did the logging get removed?

File: ana_scans.py [hunk 2]
```diff
@@ -217,13 +224,20 @@ def launchAnaArgs(anaType, cName, cType, scandate, scandatetrim=None, ztrim=4.0,
...
       pool.close()
       pool.join()
+      print "Results:", res.get()
+      for returncode in res.get():
+        if returncode != 0:
+          sys.exit(returncode)
+        pass
     except KeyboardInterrupt:
       print("Caught KeyboardInterrupt, terminating workers")
       pool.terminate()
```
Comment (jsturdy) [inline ana_scans.py]: Why no `sys.exit(-1)` here?  Or, why present in the other `except` blocks?

File: ana_scans.py (hunk 1 above)
Comment (lmoureaux) [reply ana_scans.py]: Wasn't me ;-) See bug #6

File: ana_scans.py (hunk 2 above)
Comment (lmoureaux) [reply ana_scans.py]: I used `sys.exit(-1)` to indicate an exception in `ana_scans.py`. When `ana_scans.py` works properly, the exit code of (the first failing) subcommands is just forwarded to the caller.

File: ana_scans.py (hunk 1 above)
Comment (jsturdy) [reply ana_scans.py]: OK, I guess I'll force @dorney to fix it in #23 since he introduced it originally

File: ana_scans.py (hunk 2 above)
Comment (jsturdy) [reply ana_scans.py]: OK, so you're not interested in a `-1` when you get a `KeyboardInterrupt`?

Comment (lmoureaux) [reply ana_scans.py]: Not for unit tests, but that's inconsistent behavior

File: ana_scans.py (hunk 1 above)
Comment (bdorney) [reply ana_scans.py]: Addressed in 311371684a8de283af669bf83ed0139eb4e76af5

**Comment by lmoureaux** [comment]
No, as I wouldn't be able to test it.

**Pull request merged**